src/cis/fam_cis_direct.cpp

```diff
@@ -1476,10 +1476,29 @@ configFileParams Fam_CIS_Direct::get_config_info(std::string filename) {
         try {
             std::vector<std::string> temp = info->get_value_list("memsrv_list");
             ostringstream memsrvList;
-
-            for (auto item : temp)
-                memsrvList << item << ",";
-
+            std::vector<std::string> memsrvId;
+            for (auto item : temp) {
+                std::string delim = ":";
+
+                auto start = 0U;
+                auto end = item.find(delim);
+                std::string currMemsrvId = item.substr(start, end - start);
+                if (std::find(memsrvId.begin(), memsrvId.end(), currMemsrvId) ==
+                    memsrvId.end()) {
+                    /* memsrvId does not contain currMemsrvId */
+                    memsrvId.push_back(currMemsrvId);
+                    memsrvList << item << ",";
+
+                } else {
+                    /* memsrvId contains currMemsrvId  ie; duplicate memory
+                     * server id*/
+                    ostringstream message;
+                    message << "CIS:Duplicate memory server id! ";
```

Comment:
Please add the duplicate memory server id in the message.  you can rephrase the message as

"Duplicate memory server id <memory server id> specified in fam config option memsrv_list" 

No need to print the message on console here, Printing of the error message  should  be handled by the  cis service.  Remove cout statement from here.